assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -206,10 +296,46 @@ const baseSelectors = {
 	 * @param {Object} state Data store's state.
 	 * @return {(Object|undefined)} Modules available on the site.
 	 */
-	getModules( state ) {
+	getModules: createRegistrySelector( ( select ) => ( state ) => {
 		const { modules } = state;
-
+		const registryKey = select( STORE_NAME ).getModuleRegistryKey();
+		if ( undefined !== modules ) {
+			const sortedModules = sortObjectMapByKey( modules, 'order' );
+			const mappedModules = Object.values( sortedModules ).map( ( module ) => {
+				const moduleWithComponent = { ...module };
+				if ( ModuleComponents[ registryKey ] ) {
+					// If there is a settingsComponent that was passed use it, otherwise set to the default.
+					if ( ModuleComponents[ registryKey ][ module.slug ] ) {
+						moduleWithComponent.settingsComponent = ModuleComponents[ registryKey ][ module.slug ];
+					} else {
+						moduleWithComponent.settingsComponent = DefaultModuleSettings;
+					}
+				}
+
+				return moduleWithComponent;
+			} );
+			return convertArrayListToKeyedObjectMap( mappedModules, 'slug' );
+		}
 		return modules;
+	} ),
+
+	/**
+	 * Returns the registry key being used for this registry's modules.
+	 *
+	 * We key each registry with an Integer, so we don't share registered widgets
+	 * between registries. This allows us to access the appropriate registry global
+	 * from inside selectors.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(number|undefined)} The registry for the modules store.
+	 */
+	getModuleRegistryKey( state ) {
+		const { registryKey } = state;
+
+		return registryKey;
```

Comment:
Instead of duplicating the functionality from `getWidgetRegistryKey` (which is essentially what this does), why not remove this selector/related actions and remove `getWidgetRegistryKey`—then combine them into one selector likr `getRegistryKey`. The point of the `getWidgetRegistryKey` code was to allow each registry to key data in globals. There's no reason to create _multiple_ registry keys across different sections of the app/datastore, so let's combine them instead of duplicating the functionality.

It's probably reasonable to move that code to `core/site` or something quite "core".

Sorry for not highlighting that in the IB, I can see how it was a bit vague and implied copying the existing implementation. I might've even thought that at first, but seeing as they're doing the same thing, let's consolidate these functions 🙂